packages/core/src/transforms/resize.ts

```diff
@@ -96,18 +123,21 @@ export const resize: TransformFactory<ResizeOptions> = (config, context) => {
       )
     }
 
+    finalWidth = Math.round(finalWidth)
+    finalHeight = Math.round(finalHeight)
+
     setMetadata(image, 'height', finalHeight)
     setMetadata(image, 'width', finalWidth)
     setMetadata(image, 'aspect', finalAspect)
     setMetadata(image, 'withoutEnlargement', withoutEnlargement)
     setMetadata(image, 'withoutReduction', withoutReduction)
 
     return image.resize({
-      width: Math.round(finalWidth) || undefined,
-      height: Math.round(finalHeight) || undefined,
+      width: finalWidth || undefined,
+      height: finalHeight || undefined,
       withoutEnlargement: withoutEnlargement,
       withoutReduction: withoutReduction,
-      fit: getFit(config, image),
+      fit: fit,
```

Comment:
this can be written as:
```suggestion
      fit,
```